lib/utils/compose_ts.ts

```diff
@@ -384,20 +384,6 @@ async function validateSpecifiedDockerfile(
 		`);
 	}
 
-	const { dir, ext, name } = path.parse(fullDockerfilePath);
-	if (ext) {
-		const nativePathMinusExt = path.join(dir, name);
-		if (await fs.exists(nativePathMinusExt)) {
-			throw new ExpectedError(stripIndent`
-				Error: "${name}" exists on the same folder as "${nativeDockerfilePath}".
-				When an alternative Dockerfile name is specified, a file with the same base name
-				(minus the file extension) must not exist in the same folder. This is because
-				the base name file will be auto generated and added to the tar stream that is
-				sent to balenaEngine or the Docker daemon, resulting in duplicate Dockerfiles
-				and undefined behavior.
-			`);
-		}
-	}
```

Comment:
This project validation block was deleted because it regarded the fact that a specified dockerfile like `Dockerfile.debug` could become `Dockerfile` in the tar stream that gets sent to the Docker daemon. `docker-qemu-transpose` version 1.0.3 (depends-on PR balena-io-modules/docker-qemu-transpose/pull/28) changes that. (There's the related but different case of `Dockerfile.template` becoming `Dockerfile`, but this is not covered by this validation check which is specific to specified dockerfiles.)